JWBestPracticeApps/JWVUDRMFairPlay/JWVUDRMFairPlayViewController.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  JWFairPlayDrmViewController.swift
+//  JWBestPracticeApps
+//
+//  Created by Karim Mourra on 9/26/16.
+//  Copyright © 2016 Karim Mourra. All rights reserved.
+//
+
+import vudrmFairPlaySDK
+
+
+class JWVUDRMFairPlayViewController: JWBasicVideoViewController, JWDrmDataSource, URLSessionDelegate {
+
+    let encryptedFile = "INSERT_CONTENT_URL"
+    let vudrmToken = "INSERT_TOKEN"
+    var drm: vudrmFairPlaySDK?
+    var contentID: String?
+    var licenseUrl: String?
+    var parsedAssetID: NSURL?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.player.drmDataSource = self
+    }
+    
+    override func onReady(_ event: JWEvent & JWReadyEvent) {
+        let item = JWPlaylistItem()
+        item.file = encryptedFile
+        self.player.load([item])
+    }
+    
+    func fetchAppIdentifier(forRequest loadingRequestURL: URL, for encryption: JWEncryption, withCompletion completion: @escaping (Data) -> Void) {
+        if encryption == .fairPlay {
+            // obtain application Id / Application Certificate here.
+            do {
+                
+                self.drm = vudrmFairPlaySDK()
+                let applicationCertificate = try drm?.requestApplicationCertificate(token: vudrmToken, contentID: contentID!)
+                
+                let applicationId: Data = applicationCertificate!
+                
+                completion(applicationId)
+            } catch {
+                let message = ["fetchAppIdentifier": "Unexpected error: \(error)."]
+                print(message)
+            }
+        }
+    }
+    
+    func fetchContentIdentifier(forRequest loadingRequestURL: URL, for encryption: JWEncryption, withCompletion completion: @escaping (Data) -> Void) {
+        if encryption == .fairPlay {
+            // obtain asset Id / Content Key ID here.
+            self.getAssetID(videoUrl: NSURL(string: encryptedFile)!){
+                let assetId: String = (self.parsedAssetID?.absoluteString)!
+                self.contentID = (self.parsedAssetID?.lastPathComponent)!
+                self.licenseUrl = assetId.replacingOccurrences(of: "skd", with: "https")
+                let assetIdData = Data(bytes: assetId.cString(using: .utf8)!,
+                                       count: assetId.lengthOfBytes(using: .utf8))
+                completion(assetIdData)
+            }
+        }
+    }
+    
+    func fetchContentKey(withRequest requestBytes: Data, for encryption: JWEncryption, withCompletion completion: @escaping (Data, Date?, String?) -> Void) {
+        if encryption == .fairPlay {
```

Comment:
A better way to do this is use a `guard` instead of an `if`. Same for your other methods above.

Reply:
I am happy to change this, but, the other DRM target `JWFairPlayDrm` would need to be changed as well, and the documented implementation would also need to be amended, see: [docs](https://developer.jwplayer.com/jwplayer/docs/ios-play-drm-protected-content-v3)